src/controllers/apps/ecommerce/coupon.controllers.js

```diff
@@ -19,6 +19,22 @@ const createCoupon = asyncHandler(async (req, res) => {
     expiryDate,
   } = req.body;
 
+  // Convert the provided start and expiry dates to JavaScript Date objects
+  const start = new Date(startDate);
+  const expiry = new Date(expiryDate);
+
+  // Check if the start date is in the future
+  // If the start date is in the past (i.e., before the current date and time), throw an error
+  if (start < new Date()) {
```

Comment:
There is no need to validate the start date, it can be in the past as well as the future